transcendence-app/src/chat/chatroom/chatroom-member/infrastructure/db/postgres/chatroom-member.postgres.repository.ts

```diff
@@ -115,38 +116,49 @@ export class ChatroomMemberPostgresRepository
     userId: string,
   ): Promise<ChatroomMember | null> {
     const chatroomMembersData = await makeQuery(this.pool, {
-      text: `DELETE FROM ${this.table}
-      WHERE ${ChatroomMemberKeys.CHATID} = $1
-        AND ${ChatroomMemberKeys.USERID} = $2
-      RETURNING *;`,
+      text: `DELETE
+             FROM ${this.table}
+             WHERE ${ChatroomMemberKeys.CHATID} = $1
+               AND ${ChatroomMemberKeys.USERID} = $2
+             RETURNING *;`,
       values: [chatroomId, userId],
     });
     return chatroomMembersData && chatroomMembersData.length
       ? new this.ctor(chatroomMembersData[0])
       : null;
   }
 
-  async retrieveChatroomMembers(
+  async getPaginatedChatroomMembers(
     chatroomId: string,
+    paginationQueryDto: Required<PaginationWithSearchQueryDto>,
   ): Promise<ChatroomMemberWithUser[] | null> {
+    const { limit, offset, sort, search } = paginationQueryDto;
+
+    const orderBy =
+      sort === BooleanString.True
+        ? ChatroomMemberKeys.JOINED_AT
+        : userKeys.USERNAME;
     const users = await makeQuery<ChatroomMemberWithUser>(this.pool, {
-      text: `SELECT u."username",
-                    u."avatarId",
-                    u."avatarX",
-                    u."avatarY",
-                    c."ownerId" IS NOT NULL as owner,
-                    cm."admin",
-                    cm."muted",
-                    cm."banned"
+      text: `SELECT u.${userKeys.USERNAME},
+                    u.${userKeys.AVATAR_ID},
+                    u.${userKeys.AVATAR_X},
+                    u.${userKeys.AVATAR_Y},
+                    c.${ChatroomKeys.OWNERID} IS NOT NULL as owner,
+                    cm.${ChatroomMemberKeys.ADMIN},
+                    cm.${ChatroomMemberKeys.MUTED},
+                    cm.${ChatroomMemberKeys.BANNED}
              FROM ${table.USERS} u
                     INNER JOIN ${this.table} cm
-                               ON cm."userId" = u."id"
+                               ON cm.${ChatroomMemberKeys.USERID} = u.${userKeys.ID}
                     LEFT JOIN ${table.CHATROOM} c
-                              ON c."id" = cm."chatId"
-                                AND u."id" = c."ownerId"
-             WHERE cm."chatId" = $1
-               AND cm."joinedAt" IS NOT NULL`,
-      values: [chatroomId],
+                              ON c.${ChatroomKeys.ID} = cm.${ChatroomMemberKeys.CHATID}
+                                AND u.${userKeys.ID} = c.${ChatroomKeys.OWNERID}
+             WHERE cm.${ChatroomMemberKeys.CHATID} = $1
+               AND cm.${ChatroomMemberKeys.JOINED_AT} IS NOT NULL
+               AND ${userKeys.USERNAME} ILIKE $2
+             ORDER BY ${orderBy} DESC
```

Comment:
question: What do you think about removing the `DESC`? Unless we want to show the results in reverse alphabetical order?

Reply:
We have by default sort = true. With this we're actually sorting by joinedAt date.